File: src/algorithms/cpp/Power_of_Four.cpp
#include <iostream>
#include <cstdio>
#include <cstring>
#include <cmath>
#include <cstdlib>
#include <algorithm>
#include <vector>
#include <map>
#include <set>
using namespace std;
#define out(v) cerr << #v << ": " << (v) << endl
#define SZ(v) ((int)(v).size())
const int maxint = -1u>>1;
template <class T> bool get_max(T& a, const T &b) {return b > a? a = b, 1: 0;}
template <class T> bool get_min(T& a, const T &b) {return b < a? a = b, 1: 0;}


class Solution {
public:
    bool isPowerOfFour(int num) {
        if (num <= 0)           return false;
        if ((num & (num - 1)) == 0 && (num - 1) % 3 == 0)    return true;
        else                                                 return false;
    }
};


int main() {
    return 0;
}

File: src/algorithms/python/Power_of_Four.py
#!/bin/env python2
# -*- coding: utf-8 -*-


class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        _num = num - 1
        return True if num > 0 and num & _num == 0 and _num % 3 == 0 else False
